app/controllers/map_controller.rb

```diff
@@ -1,22 +1,20 @@
 class MapController < ApplicationController
   def index
     @title = 'Maps'
-    @pagy, @nodes = pagy(Node.order('nid DESC')
-      .where(type: 'map', status: 1), items: 32)
+    nodes = Node.find_by_tag('map-archive')
+      .order('nodes.nid DESC')
+      .where(status: 1)
+    @pagy, @nodes = pagy(nodes, items: 32)
 
     @map_lat = nil
     @map_lon = nil
     if current_user&.has_power_tag("lat") && current_user&.has_power_tag("lon")
       @map_lat = current_user.get_value_of_power_tag("lat").to_f
       @map_lon = current_user.get_value_of_power_tag("lon").to_f
     end
-    # I'm not sure if this is actually eager loading the tags...
     @maps = Node.joins(:tag)
-      .where('type = "map" AND status = 1 AND (term_data.name LIKE ? OR term_data.name LIKE ?)', 'lat:%', 'lon:%')
+      .where('(term_data.name LIKE ? OR term_data.name LIKE ?) AND nid IN (?)', 'lat:%', 'lon:%', nodes.collect(&:nid))
```

Comment:
Here, there is an ambiguity error because of ```nid```. Can be fixed like this
```suggestion
      .where('(term_data.name LIKE ? OR term_data.name LIKE ?) AND node.nid IN (?)', 'lat:%', 'lon:%', nodes.collect(&:nid))
```